pkg/features/pilot/pilot.go

```diff
@@ -172,6 +172,14 @@ var (
 		"PILOT_ENABLE_REDIS_FILTER",
 		false,
 		"EnableRedisFilter enables injection of `envoy.filters.network.redis_proxy` in the filter chain.")
+
+	// UseRemoteAddress sets useRemoteAddress to true for side car outbound listeners so that it picks up the localhost
+	// address of the sender, which is an internal address, so that trusted headers are not sanitized.
+	UseRemoteAddress = useRemoteAddress.Get
+	useRemoteAddress = env.RegisterBoolVar(
+		"PILOT_USE_REMOTE_ADDRESS",
+		false,
+		"seRemoteAddress sets useRemoteAddress to true for side car outbound listeners.")
```

Comment:
I understand this is for all http connection manager. No matter inbound or outbound.

Reply:
No. I have changed it only for sidecar outbound listeners in this PR https://github.com/istio/istio/pull/14097. Now since it is creating problem in some usecases, I have guarded it with a feature flag. Other inbound listeners will continue to work as is.